src/guiguts/tools/pptxt.py

```diff
@@ -0,0 +1,6738 @@
+                if reportme:
+                    heading = "\nmixed case within word"
+                    output_record = make_a_report_record(line_index + 1, line)
+
+                    # Highlight the word within the output record just generated.
+
+                    if res := regex_search(word, output_record):
+                        output_record = decorate(
+                            output_record, [(res.start(), len(word))], "redonyellow"
+                        )
+                        if heading in specials_report:
+                            specials_report[heading].append(output_record)
+                        else:
+                            specials_report[heading] = [output_record]
+
+                if len(word) > 2:
+                    # Check word endings (last 2 characters)
+                    last2 = word[len(word) - 2 :]
+                    if res := re0003c.match(last2):
+                        template = "\nquery word ending with {}"
+                        heading = template.format(last2)
+                        output_record = make_a_report_record(line_index + 1, line)
+                        # Find the suspect word on the output record and highlight it.
+                        if res := regex_search(
+                            word, output_record
+                        ):  # Belt-n-braces. Should always match.
+                            output_record = decorate(
+                                output_record, [(res.start(), len(word))], "redonyellow"
+                            )
+                            if heading in specials_report:
+                                specials_report[heading].append(output_record)
+                            else:
+                                specials_report[heading] = [output_record]
+                    # Check word start (first 2 characters)
+                    first2 = word[0:2]
+                    if res := re0003d.match(first2):
+                        template = "\nquery word starting with {}"
+                        heading = template.format(first2)
+                        output_record = make_a_report_record(line_index + 1, line)
+                        # Find the suspect word on the output record and highlight it.
+                        if res := regex_search(
+                            word, output_record
+                        ):  # Belt-n-braces. Should always match.
+                            output_record = decorate(
+                                output_record, [(res.start(), len(word))], "redonyellow"
+                            )
+                            if heading in specials_report:
+                                specials_report[heading].append(output_record)
+                            else:
+                                specials_report[heading] = [output_record]
+
+            ##
+            # We're back to testing whole line again.
+            ##
+
+            if res := re0006.search(line):
```

Comment:
Actually, it would probably be even clearer if the regexes were defined here alongside the headers because the variable names don't make a lot of sense anyway and swap the order:
```python
line_regexes = [
    ["\nsingle character line", ["^.$"],
    ["\nbroken hyphenation", [r"(\p{L}\- +?\p{L})|(\p{L} +?\-\p{L})"]],
    ["\ncomma spacing", [r"\p{L},\p{L}", r"\p{L},\p{N}", r"\s,", r"^,"]],
    ...
]
```
Then just use them as `re.search(regex, line)` in the function. The `re` engine caches regexes and precompiling them probably isn't saving us much perf-wise.